ambari-server/src/main/python/ambari_server/setupSso.py

```diff
@@ -47,6 +47,8 @@
 JWT_PUBLIC_KEY_HEADER = "-----BEGIN CERTIFICATE-----\n"
 JWT_PUBLIC_KEY_FOOTER = "\n-----END CERTIFICATE-----\n"
 
+REGEX_URL = "http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+"
```

Comment:
This regular expression seems to be extremely lenient and I am not sure it it really doing all that much to validate a true URL other than ensure it begins with http.  Maybe just skip the URL validation or change it to something like 

```
https?:\/\/.*
```
I think we can go really crazy here validating the URL but the regular expression will be huge and unreadable.  The host side of the equation is huge alone:

```
(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])
```
_From https://stackoverflow.com/questions/106179/regular-expression-to-match-dns-hostname-or-ip-address_

Bu the user can choose to use an IP address as well, so the expression will be even larger just to validate the host part of this.

Reply:
I used the regex found on this page: http://www.noah.org/wiki/RegEx_Python#URL_regex_pattern

I'd not drop URL validation in the tool, instead we may consider using django's `URLValidator` RE:
```
regex = re.compile(
    r'^(?:http|ftp)s?://' # http:// or https://
    r'(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|' # domain...
    r'localhost|' # localhost...
    r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}|' # ...or ipv4
    r'\[?[A-F0-9]*:[A-F0-9:]+\]?)' # ...or ipv6
    r'(?::\d+)?' # optional port
    r'(?:/?|[/?]\S+)$', re.IGNORECASE)
```
Or, I'm not sure if importing 3rd party modules is an option, we may use the django lib.
Further info is available here: https://github.com/django/django/blob/master/django/core/validators.py

What do you think?

Reply:
ok.. well then leave what you have since others seem to think it is a useful Regex for URLs.